modules/ting_field_search/ting_field_search.install

```diff
@@ -0,0 +1,354 @@
+    'unique keys' => array(
+      'name' => array('name'),
+    ),
+  );
+
+  // For performance reasons keep facet settings in a seperate table.
+  // TODO: Consider moving these settings to the new config field.
+  $schema['ting_field_search_facet'] = array(
+    'description' => 'Holds facet settings for each profile.',
+    'fields' => array(
+      'name' => array(
+        'type' => 'varchar',
+        'length' => 64,
+        'not null' => TRUE,
+        'default' => '',
+        'description' => 'Facet name: Name of the facet (e.g. facet.type).',
+      ),
+      'pid' => array(
+        'type' => 'int',
+        'not null' => TRUE,
+        'unsigned' => TRUE,
+        'description' => 'Foreign key: ID of the associated profile.',
+        'default' => 0,
+      ),
+      'title' => array(
+        'type' => 'varchar',
+        'length' => 64,
+        'not null' => TRUE,
+        'default' => '',
+        'description' => 'Facet title: The title of the facet, as shown to users in UI',
+      ),
+      'sorting' => array(
+        'type' => 'varchar',
+        'length' => 32,
+        'not null' => TRUE,
+        'default' => 'default',
+        'description' => 'Facet sorting. Sorting of the terms in the facet.',
+      ),
+      'weight' => array(
+        'type' => 'int',
+        'size' => 'small',
+        'not null' => TRUE,
+        'default' => 1,
+        'description' => 'Facet weight: Used to order facets in the UI.'
+      ),
+    ),
+    'primary key' => array('name', 'pid'),
+    'foreign keys' => array(
+      'profile' => array(
+        'table' => 'ting_field_search_profile',
+        'columns' => array('profile' => 'pid'),
+      ),
+    ),
+  );
+
+  return $schema;
+}
+
+/**
+ * Implements hook_requirements().
+ */
+function ting_field_search_requirements($phase) {
+  $requirements = array();
+
+  if ($phase != 'runtime') {
+    return $requirements;
+  }
+
+  if (!ting_field_search_get_status()) {
+    $requirements['ting_field_search'] = array(
+      'title' => t('Ting field search'),
+      'value' => t('Ting field search is NOT installed correctly and will not work proberly. Please go to <a href="@install">Installation</a> to correct this.', array(
+          '@install' => url('admin/config/ting-field-search/install', array(
+            'query' => drupal_get_destination()
+          )),
+        )
+      ),
+      'severity' => REQUIREMENT_ERROR,
+    );
+  }
+
+  return $requirements;
+}
+
+/**
+ * Helper function to load profiles in update functions.
+ * For schema version 7000 -> ...
+ */
+function _ting_field_search_load_profiles_7000() {
+  return db_select('ting_field_search_profile', 'tp')
+    ->fields('tp')
+    ->execute()
+    ->fetchAllAssoc('name');
+}
+
+/**
+ * Adds the new config field to the profile table.
+ * Adds new serial primary key to profile table.
+ * Update foreign key in facet table to point to the new profile primary key.
+ */
+function ting_field_search_update_7000() {
```

Comment:
Technically these update functions are not run on DDB sites as this is a new module? So should they be removed?